straxen/plugins/peaks/peaks_subtyping.py

```diff
@@ -0,0 +1,264 @@
+        2. extend window after pS2
+        3. mark all other large S2s as olS2. 
+        4. mark all other small S2s as photoionization (PH)
+
+        if pS2 identified, extend the window to 2fdt after such pS2 and mark S2s:
+        1. other large S2 if an S2 is larger than ls2_threshold and half the pS2 size
+        2. photoionization if not olS2
+        '''
+
+        dmask = mask[mask == 6]
+        ts = p_t[mask == 6] #time
+        tps = p_type[mask == 6] #type
+        pas = p_a[mask == 6] #area
+        s1s_t = ts[tps == 1] # s1 time
+        s1s_a = pas[tps == 1] # s1 area
+        s2s_t = ts[tps == 2] # s2 time
+        s2s_a = pas[tps == 2] # s2 area
+        s1_mask = dmask[tps == 1] # define s1 subtypes
+        s2_mask = dmask[tps == 2] # define s2 subtypes
+
+        for i in range(len(s1s_t)):
+            # locate pS2:
+            ps2_cand_cond = (s2s_t > s1s_t[i]) & (s2s_t <= s1s_t[i] + s1_s2_window)
+            if len(s2s_a[ps2_cand_cond]):
+                # do we have pS2 or a bunch of small S2s
+                max_ind,max_val = self.find_max_ind(s2s_a[ps2_cand_cond])
+                max_time = s2s_t[ps2_cand_cond][max_ind]
+                if max_val >= ls2_threshold:
+                    # pS2 located
+                    whole_window_cond = (s2s_t > s1s_t[i])
+                    whole_window_cond &= (s2s_t <= s2s_t[ps2_cand_cond][max_ind] + 2 * full_dt)
+                    s2_mask_tmp = s2_mask[whole_window_cond]
+                    s2_mask_tmp[
+                        (s2s_a[whole_window_cond] >= np.max([ls2_threshold,other_ls2 * max_val]))
+                        & (s2_mask[whole_window_cond] != 22)
+                        & (s2s_t[whole_window_cond] < max_time)] = 261
+                    s2_mask_tmp[
+                        (s2s_a[whole_window_cond] >= np.max([ls2_threshold,other_ls2 * max_val]))
+                        & (s2_mask[whole_window_cond] != 22)
+                        & (s2s_t[whole_window_cond] >= max_time)] = 262
+                    s2_mask_tmp[
+                        (s2s_a[whole_window_cond] < np.max([ls2_threshold,other_ls2 * max_val]))
+                        & (s2_mask[whole_window_cond] != 22)
+                        & (s2s_t[whole_window_cond] < max_time)] = 271
+                    s2_mask_tmp[
+                        (s2s_a[whole_window_cond] < np.max([ls2_threshold,other_ls2 * max_val]))
+                        & (s2_mask[whole_window_cond] != 22)
+                        & (s2s_t[whole_window_cond] >= max_time)] = 272
+                    # assign the pS2 and S1
+                    s2_mask_tmp[s2s_t[whole_window_cond] == max_time] = 22
+                    s1_mask[i] = 11
+                    s2_mask[whole_window_cond] = s2_mask_tmp
+                else:
+                    #no pS2, loneS1 spotted
+                    whole_window_cond = (s2s_t > s1s_t[i]) & (s2s_t <= s1s_t[i] + full_dt)
```

Comment:
Is there a potential bug? All other time window length is defined as `2 * full_dt`.